multicluster/cmd/multicluster-controller/member_test.go

```diff
@@ -0,0 +1,123 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main under directory cmd parses and validates user input,
+// instantiates and initializes objects imported from pkg, and runs
+// the process.
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	k8smcsv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	mcsv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/test/mocks"
+)
+
+var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+
+// The origin signal handler function will close a channel twice. If the runLeader and runMember run at the same time,
+// there will be a panic. So we need another signal handler function to avoid it.
+func mockSetupSignalHandler() context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	c := make(chan os.Signal, 2)
+	signal.Notify(c, shutdownSignals...)
+	go func() {
+		<-c
+		cancel()
+		<-c
+		os.Exit(1)
+	}()
+
+	return ctx
+}
+
+var mockMemberManager *mocks.MockManager
+
+func initMemberMockManager(mockCtrl *gomock.Controller) {
+	newScheme := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(newScheme))
+	utilruntime.Must(k8smcsv1alpha1.AddToScheme(newScheme))
+	utilruntime.Must(mcsv1alpha1.AddToScheme(newScheme))
+	fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects().Build()
+
+	mockMemberManager = mocks.NewMockManager(mockCtrl)
+	mockMemberManager.EXPECT().GetWebhookServer().Return(&webhook.Server{}).AnyTimes()
+	mockMemberManager.EXPECT().GetClient().Return(fakeClient).AnyTimes()
+	mockMemberManager.EXPECT().GetScheme().Return(newScheme).AnyTimes()
+	mockMemberManager.EXPECT().GetControllerOptions().Return(v1alpha1.ControllerConfigurationSpec{}).AnyTimes()
+	mockMemberManager.EXPECT().GetLogger().Return(klog.NewKlogr()).AnyTimes()
+	mockMemberManager.EXPECT().SetFields(gomock.Any()).Return(nil).AnyTimes()
+	mockMemberManager.EXPECT().Add(gomock.Any()).Return(nil).AnyTimes()
+	mockLeaderManager.EXPECT().GetConfig().Return(&rest.Config{}).AnyTimes()
+	mockMemberManager.EXPECT().Start(gomock.Any()).Return(nil).AnyTimes()
+}
```

Comment:
Could you check if `initMemberMockManager` and `initLeaderMockManager` can be one `initMockManager`?